pkgs/development/python-modules/rsa/default.nix

```diff
@@ -1,36 +1,49 @@
 { lib
 , buildPythonPackage
-, fetchPypi
-, unittest2
+, fetchFromGitHub
+, mypy
+, poetry-core
 , pyasn1
-, mock
-, isPy3k
+, pytestCheckHook
 , pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "rsa";
   version = "4.9";
+  format = "pyproject";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-44RkpJxshdfxNRsBJmYUh6fgoUpQ8WdexQ6zTU8g7yE=";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "sybrenstuvel";
+    repo = "python-rsa";
+    rev = "version-${version}";
+    hash = "sha256-PwaRe+ICy0UoguXSMSh3PFl5R+YAhJwNdNN9isadlJY=";
   };
 
-  checkInputs = [ unittest2 mock ];
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
   propagatedBuildInputs = [ pyasn1 ];
 
-  preConfigure = lib.optionalString (isPy3k && pythonOlder "3.7") ''
-    substituteInPlace setup.py --replace "open('README.md')" "open('README.md',encoding='utf-8')"
+  preCheck = ''
+    sed -i '/addopts/d' tox.ini
   '';
 
-  # No tests in archive
-  doCheck = false;
+  checkInputs = [
+    mypy
```

Comment:
Can we easily skip typechecks?

Reply:
Yes, it's possible. I disabled `tests/test_mypy.py`.